Adding MaxPro documentation and tutorials

sotorrio1 opened the pull request:
## Fixes/Addresses:

Issue #1269 

## Summary/Motivation:
The newly added SDoE capability, MaxPro, needs documentation and tutorials to help users better understand the theory behind it and how to use the tool to generate MaxPro designs. 

## Changes proposed in this PR:
- Adding a section about USF available methods to explain MaxPro in detail. 
- Adding two tutorials showing how to generate a simple MaxPro design, and an augmented one. 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the copyright and license terms described in the LICENSE.md file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

henry-gatech reviewed (dismissed):
Checked the detailed steps of the initial and the sequential designs to be correct. Verified the plots and the mathematical expressions of the USF methods to be correct. Verified the screenshot plots of utilizing the MaxPro design in the software to be correct.

sotorrio1 commented:
@ksbeattie I found a few places where we were still using old naming for some parts of the SDoE module. I updated those and also the references to installing PSUADE. Furthermore, I also updated the figures in the MaxPro sections, making them easier to read. Let me know if you find anything else.

pn51 commented on docs/source/chapt_sdoe/examples-maxpro.rst:
```diff
@@ -0,0 +1,142 @@
+   MaxPro Example 2: Load Data
+
+2. Click on the **Continue** button at the bottom right of the **Design Setup** box. This will activate the
+**Design Construction** box. By clicking on the **View** button under the **Visualize** column, user can see both the
```
the user
